src/engraving/dom/engravingitem.cpp

```diff
@@ -1198,6 +1233,114 @@ bool EngravingItem::setProperty(Pid propertyId, const PropertyValue& v)
+                linkedItem->score()->deselect(linkedItem);
+            }
+            linkedItem->score()->undoRemoveElement(linkedItem, false);
+            linkedItem->undoUnlink();
+        }
+    } else {
+        score()->undoAddElement(this, /*addToLinkedStaves*/ true, /*ctrlModifier*/ false, this);
+    }
+}
+
+void EngravingItem::relinkPropertiesToMaster(PropertyGroup propertyGroup)
+{
+    assert(!score()->isMaster());
+
+    std::list<EngravingObject*> linkedElements = linkListForPropertyPropagation();
+    EngravingObject* masterElement = nullptr;
+    for (EngravingObject* element : linkedElements) {
+        if (element->score()->isMaster()) {
+            masterElement = element;
+            break;
+        }
+    }
+
+    if (!masterElement) {
+        return;
+    }
+
+    const std::set<Pid>& propertiesToRelink = propertyGroup == PropertyGroup::POSITION ? positionProperties()
+                                              : propertyGroup == PropertyGroup::APPEARANCE ? appearanceProperties()
+                                              : textProperties();
+
+    for (Pid propertyId : propertiesToRelink) {
+        PropertyValue masterValue = masterElement->getProperty(propertyId);
+        PropertyFlags masterFlags = masterElement->propertyFlags(propertyId);
+        setProperty(propertyId, masterValue);
+        setPropertyFlags(propertyId, masterFlags);
+    }
+
+    return;
+}
+
+PropertyPropagation EngravingItem::propertyPropagation(EngravingItem* destinationItem, Pid propertyId)
+{
+    assert(destinationItem != this);
+
+    if (propertyLink(propertyId)) {
+        return PropertyPropagation::PROPAGATE; // These properties are always linked, no matter what
+    }
+
+    static std::set<Pid> NEVER_PROPAGATING_PROPERTIES = {
+        Pid::POSITION_LINKED_TO_MASTER,
+        Pid::APPEARANCE_LINKED_TO_MASTER,
+        Pid::TEXT_LINKED_TO_MASTER,
+        Pid::GENERATED,
+        Pid::EXCLUDE_FROM_OTHER_PARTS
+    };
+
+    if (mu::contains(NEVER_PROPAGATING_PROPERTIES, propertyId)) {
+        return PropertyPropagation::NONE;
+    }
+
+    Score* sourceScore = score();
+    Score* destinationScore = destinationItem->score();
+    bool isTextProperty = mu::contains(textProperties(), propertyId);
+
+    if ((isTextProperty && isPropertyLinkedToMaster(propertyId)) || sourceScore == destinationScore) {
+        return PropertyPropagation::PROPAGATE;
+    }
+
+    if (sourceScore != destinationScore && !sourceScore->isMaster()) {
+        // Properties are only propagated when being edited from master. If this is being edited
+        // from a part score, we mark it as unlinked so it becomes independent in the part.
+        return PropertyPropagation::UNLINK;
+    }
+
+    if (destinationItem->isPropertyLinkedToMaster(propertyId) || sourceScore == destinationScore) {
+        return PropertyPropagation::PROPAGATE;
+    }
+
+    return PropertyPropagation::NONE;
+}
+
+bool EngravingItem::canBeExcludedFromOtherParts() const
```

Comment:
Is there a specific reason to choose for this implementation instead of making this a virtual method? I consider it closely related to `manageExclusionFromParts`, which _is_ virtual, so I'd personally prefer to keep these methods close to each other.

Reply:
Mostly because for such simple methods I hate to spread the information across 10 different files: it makes it harder later to find what I need, for no real benefit. I agree that in this case it looks more consistent with manageExclusionFromParts though